### DISCUSSION TITLE
Issues with xformers version 0.0.16rc425?

### DISCUSSION THREAD

**Discussion by Evil-Dragon** [d8ahazard/sd_dreambooth_extension#857] [Q&A] [answered]
Has anyone had any success training with the xformers version now installed by A111? It was causing my training to either not work at all by producing samples that didn't take any of the training images in or cause NaN loss values despite settings or LR values. Since switching to flash_attention it now works and trains properly but obviously it's a lot slower.

Just wondering if anyone else can confirm or if it's just me?

**Comment by Zuxier** [comment, marked as answer]
yeah, new xformer version doesn't work, join the discord if you can there is a link in the wiki.

**Reply by Evil-Dragon** [reply]
Glad it's not just me then, I found the older version of xformers from https://github.com/C43H66N12O12S2/stable-diffusion-webui/releases/download/torch13/xformers-0.0.14.dev0-cp310-cp310-win_amd64.whl so i'm hoping I can just roll back to that version for the meantime.

**Reply by Zuxier** [reply]
even cu117 is scuffed

**Reply by Zuxier** [reply]
`pip uninstall torch torchvision`
`pip uninstall torchaudio`
`pip uninstall xformers`
`pip install torch torchvision --extra-index-url https://download.pytorch.org/whl/cu116`
`pip install -U -I --no-deps https://github.com/C43H66N12O12S2/stable-diffusion-webui/releases/download/torch13/xformers-0.0.14.dev0-cp310-cp310-win_amd64.whl`

be sure to be with the venv active

**Reply by FugueSegue** [reply]
I've read elsewhere that those commands should be executed within a Scripts directory that is within the venv directory. I do not see any Scripts folder in stable-diffusion-webui/venv

**Comment by flowerdealer** [comment]
Oh, so that is what it it is.

**Comment by seoeaa** [comment]
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Commit hash: 645f4e7ef8c9d59deea7091a22373b2da2b780f2
Installing requirements for Web UI

#######################################################################################################
Initializing Dreambooth
If submitting an issue on github, please provide the below text for debugging purposes:

Python revision: 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Dreambooth revision: 9f4d931a319056c537d24669cb950d146d1537b0
SD-WebUI revision: 645f4e7ef8c9d59deea7091a22373b2da2b780f2

Checking Dreambooth requirements...
[+] bitsandbytes version 0.35.0 installed.
[+] diffusers version 0.10.2 installed.
[+] transformers version 4.25.1 installed.
[+] xformers version 0.0.16rc425 installed.
[+] torch version 1.13.1+cu117 installed.
[+] torchvision version 0.14.1+cu117 installed.


yes everything works for me
I think it was necessary to update bitsandbytes version 0.35.0 installed
I don't remember exactly what I did

motel 2.1 works
model 1.5 not working

**Comment by mart-hill** [comment]
Ohh, so that's why training tends to "die off" on SD 1.5 that quickly?

![loss_plot_3840](https://user-images.githubusercontent.com/55754730/215358672-260e893d-4828-4611-bc51-23f3c927cde7.png)

Indeed, I'm using

Python revision: 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Dreambooth revision: 9f4d931a319056c537d24669cb950d146d1537b0
SD-WebUI revision: 00dab8f10defbbda579a1bc89c8d4e972c58a20d

Checking Dreambooth requirements...
[+] bitsandbytes version 0.35.0 installed.
[+] diffusers version 0.10.2 installed.
[+] transformers version 4.25.1 installed.
[+] xformers version 0.0.16rc425 installed.
[+] torch version 1.13.1+cu117 installed.
[+] torchvision version 0.14.1+cu117 installed.
